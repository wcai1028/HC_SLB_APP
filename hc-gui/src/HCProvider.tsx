import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
