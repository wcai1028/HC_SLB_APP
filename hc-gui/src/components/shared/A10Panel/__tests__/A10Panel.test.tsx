import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import * as moxios from 'moxios'
import HCProvider from 'src/HCProvider'

const renderTestComponent = () => {
  const A10Panel = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <A10Panel />
    </HCProvider>,
  )
}
const renderTestComponentWithProps = () => {
  const A10Panel = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <A10Panel
        menu="Test page"
        isFormContent="false"
        shouldShowTitle="false"
      />
    </HCProvider>,
  )
}

const getComponentInstance = (testRenderer: any) => {
  const testInstance = testRenderer.root
  const testComponent = testInstance.find(
    (node: any) => node.type.name === 'A10Panel',
  )
  return testComponent.instance
}
beforeEach(() => {
  moxios.install()
})
afterEach(() => {
  moxios.uninstall()
})

describe('test component: A10Panel', () => {
  test('should render properly', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })
  test('should render properly with props', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponentWithProps()
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot('Props is not default')
      done()
    })
  })
})
