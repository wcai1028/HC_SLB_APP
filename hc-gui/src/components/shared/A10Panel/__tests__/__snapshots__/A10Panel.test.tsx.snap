// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: A10Panel should render properly 1`] = `
<div
  className="a10WidgetsPanel"
>
  <div
    className="a10WidgetsPanelHeader"
  >
    <div
      className="a10WidgetsPanelHeaderTitle"
    />
  </div>
  <div
    className="a10-gui-widgets-panel-body"
  />
</div>
`;

exports[`test component: A10Panel should render properly with props: Props is not default 1`] = `
<div
  className="a10WidgetsPanel"
>
  <div
    className="a10WidgetsPanelHeader"
  >
    <div
      className="a10WidgetsPanelHeaderTitle"
    />
    <div
      className="a10WidgetsPanelHeaderMenu"
    >
      Test page
    </div>
  </div>
  <div
    className="a10-gui-widgets-panel-body"
  />
</div>
`;
